### PR TITLE
add python solution for word processor (987)

### PR DESCRIPTION
**Pull request by jjliewie**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-987.mdx [hunk 1]
```diff
@@ -53,4 +53,44 @@ int main(){
...
+<PySection>
+
+## Implementation
+
+```python
+import sys
+
+sys.stdin = open("word.in", "r")
+sys.stdout = open("word.out", "w")
+input = sys.stdin.readline
```
Comment (SansPapyrus683) [inline solutions/usaco-987.mdx:65]: `with`?

File: solutions/usaco-987.mdx [hunk 2]
```diff
@@ -53,4 +53,44 @@ int main(){
...
+
+def bessie_word(n, k, t):
```
Comment (SansPapyrus683) [inline solutions/usaco-987.mdx:67]: literally no need to have this in a function

File: solutions/usaco-987.mdx [hunk 3]
```diff
@@ -53,4 +53,44 @@ int main(){
...
+
+	# split string into list
+	n_list = t.split()
+
+	result = [n_list[0]]
+	cur_idx = 0
+	word_cnt = len(result[cur_idx])
```
Comment (SansPapyrus683) [inline solutions/usaco-987.mdx:74]: (optional) please use something other than `cnt`

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-987.mdx (hunk 2 above)
Comment (jjliewie) [reply solutions/usaco-987.mdx:67]: okay ill change it

File: solutions/usaco-987.mdx (hunk 3 above)
Comment (jjliewie) [reply solutions/usaco-987.mdx:74]: but is it not word count

File: solutions/usaco-987.mdx (hunk 1 above)
Comment (jjliewie) [reply solutions/usaco-987.mdx:65]: okokokokokok im just not used to it .

File: solutions/usaco-987.mdx (hunk 2 above)
Comment (jjliewie) [reply solutions/usaco-987.mdx]: fixed

File: solutions/usaco-987.mdx (hunk 3 above)
Comment (jjliewie) [reply solutions/usaco-987.mdx]: idk just tell me what to change it to pls

File: solutions/usaco-987.mdx [hunk 4]
```diff
@@ -53,4 +53,35 @@ int main(){
...
+
+## Implementation
+
+```python
+with open('word.in') as read:
+	n, k = map(int, read.readline().split())
+	t = read.readline()
+
+# split string into list
+n_list = t.split()
```
Comment (SansPapyrus683) [inline solutions/usaco-987.mdx:66]: (optional) rename to `word_list`

File: solutions/usaco-987.mdx [hunk 5]
```diff
@@ -53,4 +53,35 @@ int main(){
...
+
+result = [n_list[0]]
+cur_idx = 0
+word_cnt = len(result[cur_idx])
```
Comment (SansPapyrus683) [inline solutions/usaco-987.mdx:70]: how about renaming it to `curr_len`

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-987.mdx [hunk 6]
```diff
@@ -53,4 +53,35 @@ int main(){
...
+	if len(n_list[i]) + word_cnt > k:
+		cur_idx += 1
+		word_cnt = len(n_list[i])
+		result.append(n_list[i])
+	else:
+		result[cur_idx] += " " + n_list[i]
+		word_cnt = len(n_list[i]) + word_cnt
+
+for i in result:
+	print(i, file=open('word.out', 'w'))
```
[the remark this answers is not in the document]
Comment (jjliewie) [reply solutions/usaco-987.mdx]: fixed

File: solutions/usaco-987.mdx (hunk 5 above)
Comment (jjliewie) [reply solutions/usaco-987.mdx]: okay

File: solutions/usaco-987.mdx [hunk 7]
```diff
@@ -53,4 +53,35 @@ int main(){
...
+	n, k = map(int, read.readline().split())
+	t = read.readline()
+
+# split string into list
+word_list = t.split()
+
+result = [word_list[0]]
+cur_idx = 0
+cur_len = len(result[cur_idx])
+for i in range(1, n):
```
Comment (ryanchou-dev) [inline solutions/usaco-987.mdx:71]: space out

**Comment by ryanchou-dev** [comment]
omg so many prs

File: solutions/usaco-987.mdx (hunk 7 above)
Comment (jjliewie) [reply solutions/usaco-987.mdx:74]: yep

Comment (SansPapyrus683) [reply solutions/usaco-987.mdx:74]: what the

Comment (jjliewie) [reply solutions/usaco-987.mdx:74]: what should i do then?

Comment (SansPapyrus683) [reply solutions/usaco-987.mdx:74]: i'd leave it as is tbh

Comment (jjliewie) [reply solutions/usaco-987.mdx:74]: okay

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]
⭐

**Pull request merged**